### ISSUE TITLE
v8::ObjectTemplate::Set() with non-primitive values is deprecated

### ISSUE THREAD

**Issue by The-Alchemist** [nebrius/raspi-serial#1]
# Specification
- Raspberry Pi 3
- Raspbian Jessie
- `$ uname -a`

```
Linux raspberrypi 4.4.14-v7+ #896 SMP Sat Jul 2 15:09:43 BST 2016 armv7l GNU/Linux
```
- 
  `$ apt show nodejs`

```
Package: nodejs
Version: 6.3.0-1nodesource1~jessie1
Maintainer: Chris Lea <chl@nodesource.com>
Installed-Size: 43.5 MB
Depends: libc6 (>= 2.16), libgcc1 (>= 1:4.4.0), libstdc++6 (>= 4.9), rlwrap
Conflicts: nodejs-dev, nodejs-legacy, npm
Replaces: nodejs-dev (<= 0.8.22), nodejs-legacy, npm (<= 1.2.14)
Provides: nodejs-dev, nodejs-legacy, npm
Homepage: http://nodejs.org/
Priority: extra
Section: web
Download-Size: 8,671 kB
APT-Manual-Installed: yes
APT-Sources: https://deb.nodesource.com/node_6.x/ jessie/main armhf Packages
Description: Node.js event-based server-side javascript engine
 Node.js is similar in design to and influenced by systems like
 Ruby's Event Machine or Python's Twisted.
 .
 It takes the event model a bit further - it presents the event
 loop as a language construct instead of as a library.
 .
 Node.js is bundled with several useful libraries to handle server tasks :
 System, Events, Standard I/O, Modules, Timers, Child Processes, POSIX,
 HTTP, Multipart Parsing, TCP, DNS, Assert, Path, URL, Query Strings.

N: There is 1 additional record. Please use the '-a' switch to see it
```
# Code

```
var raspi = require('raspi-io');
var five = require('johnny-five');
var board = new five.Board({
  io: new raspi()
});

board.on('ready', function() {

  // Create an Led on pin 7 (GPIO4) on P1 and strobe it on/off
  // Optionally set the speed; defaults to 100ms
  (new five.Led('P1-7')).strobe();

});
```
# Stack Trace

```
==== C stack trace ===============================

(empty)
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

==== JS stack trace =========================================

Security context: 0x57a78fd9 <JS Object>#0#
    1: .node [module.js:568] [pc=0x23953424] (this=0x24811189 <an Object with map 0x2d80cfbd>#1#,module=0x249b540d <a Module with map 0x2d80d2a9>#2#,filename=0x249b53a9 <String[88]: /home/pi/node_modules/raspi-serial/node_modules/serialport/build/Release/serialport.node>)
    2: load [module.js:458] [pc=0x2393c4b8] (this=0x249b540d <a Module with map 0x2d80d2a9>#2#,filename=0x249b53a9 <String[88]: /home/pi/node_modules/raspi-serial/node_modules/serialport/build/Release/serialport.node>)
    3: tryModuleLoad(aka tryModuleLoad) [module.js:417] [pc=0x2393bfa8] (this=0x57a080c9 <undefined>,module=0x249b540d <a Module with map 0x2d80d2a9>#2#,filename=0x249b53a9 <String[88]: /home/pi/node_modules/raspi-serial/node_modules/serialport/build/Release/serialport.node>)
    4: _load [module.js:409] [pc=0x23936f94] (this=0x2481121d <JS Function Module (SharedFunctionInfo 0x4141d2f1)>#3#,request=0x249b4d81 <String[88]: /home/pi/node_modules/raspi-serial/node_modules/serialport/build/Release/serialport.node>,parent=0x2490dbe5 <a Module with map 0x2d80d2a9>#4#,isMain=0x57a08161 <false>)
    5: require [module.js:468] [pc=0x23946f98] (this=0x2490dbe5 <a Module with map 0x2d80d2a9>#4#,path=0x249b4d81 <String[88]: /home/pi/node_modules/raspi-serial/node_modules/serialport/build/Release/serialport.node>)
    6: require(aka require) [internal/module.js:20] [pc=0x23946c6c] (this=0x57a080c9 <undefined>,path=0x249b4d81 <String[88]: /home/pi/node_modules/raspi-serial/node_modules/serialport/build/Release/serialport.node>)
    7: bindings [/home/pi/node_modules/bindings/bindings.js:76] [pc=0x23966294] (this=0x57a87801 <JS Global Object>#5#,opts=0x414a9039 <String[15]: serialport.node>)
    8: /* anonymous */ [/home/pi/node_modules/raspi-serial/node_modules/serialport/lib/bindings.js:3] [pc=0x23992424] (this=0x249ae191 <an Object with map 0x5cd09de5>#6#,exports=0x249ae191 <an Object with map 0x5cd09de5>#6#,require=0x249af05d <JS Function require (SharedFunctionInfo 0x414390d1)>#7#,module=0x249ae169 <a Module with map 0x2d80d2a9>#8#,__filename=0x249ae111 <String[74]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib/bindings.js>,__dirname=0x249af029 <String[62]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib>)
    9: _compile [module.js:541] [pc=0x239460bc] (this=0x249ae169 <a Module with map 0x2d80d2a9>#8#,content=0x249ae7d1 <String[784]\: 'use strict';\n\nvar bindings = require('bindings')('serialport.node');\nvar listUnix = require('./list-unix');\n\nvar linux = process.platform !== 'win32' && process.platform !== 'darwin';\n\nfunction listLinux(callback) {\n  callback = callback || function(err) {\n    if (err) { this.emit('error', err) }\n  }.bind(this);\n  return listUnix(callback);\n};\n\nvar platformOptions = {};\nif (process.platform !== 'win32') {\n  platformOptions = {\n    vmin: 1,\n    vtime: 0\n  };\n}\n\nmodule.exports = {\n  close: bindings.close,\n  drain: bindings.drain,\n  flush: bindings.flush,\n  list: linux ? listLinux : bindings.list,\n  open: bindings.open,\n  SerialportPoller: bindings.SerialportPoller,\n  set: bindings.set,\n  update: bindings.update,\n  write: bindings.write,\n  platformOptions: platformOptions\n};\n>,filename=0x249ae111 <String[74]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib/bindings.js>)
   10: .js [module.js:550] [pc=0x2393de74] (this=0x24811189 <an Object with map 0x2d80cfbd>#1#,module=0x249ae169 <a Module with map 0x2d80d2a9>#8#,filename=0x249ae111 <String[74]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib/bindings.js>)
   11: load [module.js:458] [pc=0x2393c4b8] (this=0x249ae169 <a Module with map 0x2d80d2a9>#8#,filename=0x249ae111 <String[74]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib/bindings.js>)
   12: tryModuleLoad(aka tryModuleLoad) [module.js:417] [pc=0x2393bfa8] (this=0x57a080c9 <undefined>,module=0x249ae169 <a Module with map 0x2d80d2a9>#8#,filename=0x249ae111 <String[74]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib/bindings.js>)
   13: _load [module.js:409] [pc=0x23936f94] (this=0x2481121d <JS Function Module (SharedFunctionInfo 0x4141d2f1)>#3#,request=0x41497369 <String[10]: ./bindings>,parent=0x2496c809 <a Module with map 0x2d80d2a9>#9#,isMain=0x57a08161 <false>)
   14: require [module.js:468] [pc=0x23946f98] (this=0x2496c809 <a Module with map 0x2d80d2a9>#9#,path=0x41497369 <String[10]: ./bindings>)
   15: require(aka require) [internal/module.js:20] [pc=0x23946c6c] (this=0x57a080c9 <undefined>,path=0x41497369 <String[10]: ./bindings>)
   16: /* anonymous */ [/home/pi/node_modules/raspi-serial/node_modules/serialport/lib/serialport.js:16] [pc=0x239794d4] (this=0x2496c831 <an Object with map 0x5cd09de5>#10#,exports=0x2496c831 <an Object with map 0x5cd09de5>#10#,require=0x24973f65 <JS Function require (SharedFunctionInfo 0x414390d1)>#11#,module=0x2496c809 <a Module with map 0x2d80d2a9>#9#,__filename=0x2496c7b1 <String[76]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib/serialport.js>,__dirname=0x24973f31 <String[62]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib>)
   17: _compile [module.js:541] [pc=0x239460bc] (this=0x2496c809 <a Module with map 0x2d80d2a9>#9#,content=0x2496cf19 <Very long string[13557]>#12#,filename=0x2496c7b1 <String[76]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib/serialport.js>)
   18: .js [module.js:550] [pc=0x2393de74] (this=0x24811189 <an Object with map 0x2d80cfbd>#1#,module=0x2496c809 <a Module with map 0x2d80d2a9>#9#,filename=0x2496c7b1 <String[76]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib/serialport.js>)
   19: load [module.js:458] [pc=0x2393c4b8] (this=0x2496c809 <a Module with map 0x2d80d2a9>#9#,filename=0x2496c7b1 <String[76]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib/serialport.js>)
   20: tryModuleLoad(aka tryModuleLoad) [module.js:417] [pc=0x2393bfa8] (this=0x57a080c9 <undefined>,module=0x2496c809 <a Module with map 0x2d80d2a9>#9#,filename=0x2496c7b1 <String[76]: /home/pi/node_modules/raspi-serial/node_modules/serialport/lib/serialport.js>)
   21: _load [module.js:409] [pc=0x23936f94] (this=0x2481121d <JS Function Module (SharedFunctionInfo 0x4141d2f1)>#3#,request=0x41493855 <String[10]: serialport>,parent=0x2495a571 <a Module with map 0x2d80d2a9>#13#,isMain=0x57a08161 <false>)
   22: require [module.js:468] [pc=0x23946f98] (this=0x2495a571 <a Module with map 0x2d80d2a9>#13#,path=0x41493855 <String[10]: serialport>)
   23: require(aka require) [internal/module.js:20] [pc=0x23946c6c] (this=0x57a080c9 <undefined>,path=0x41493855 <String[10]: serialport>)
   24: /* anonymous */ [/home/pi/node_modules/raspi-serial/lib/index.js:12] [pc=0x23978754] (this=0x2495a599 <an Object with map 0x2d8197f9>#14#,exports=0x2495a599 <an Object with map 0x2d8197f9>#14#,require=0x249673a1 <JS Function require (SharedFunctionInfo 0x414390d1)>#15#,module=0x2495a571 <a Module with map 0x2d80d2a9>#13#,__filename=0x2495a535 <String[47]: /home/pi/node_modules/raspi-serial/lib/index.js>,__dirname=0x2496736d <String[38]: /home/pi/node_modules/raspi-serial/lib>)
   25: _compile [module.js:541] [pc=0x239460bc] (this=0x2495a571 <a Module with map 0x2d80d2a9>#13#,content=0x2495abf5 <Very long string[25156]>#16#,filename=0x2495a535 <String[47]: /home/pi/node_modules/raspi-serial/lib/index.js>)
   26: .js [module.js:550] [pc=0x2393de74] (this=0x24811189 <an Object with map 0x2d80cfbd>#1#,module=0x2495a571 <a Module with map 0x2d80d2a9>#13#,filename=0x2495a535 <String[47]: /home/pi/node_modules/raspi-serial/lib/index.js>)
   27: load [module.js:458] [pc=0x2393c4b8] (this=0x2495a571 <a Module with map 0x2d80d2a9>#13#,filename=0x2495a535 <String[47]: /home/pi/node_modules/raspi-serial/lib/index.js>)
   28: tryModuleLoad(aka tryModuleLoad) [module.js:417] [pc=0x2393bfa8] (this=0x57a080c9 <undefined>,module=0x2495a571 <a Module with map 0x2d80d2a9>#13#,filename=0x2495a535 <String[47]: /home/pi/node_modules/raspi-serial/lib/index.js>)
   29: _load [module.js:409] [pc=0x23936f94] (this=0x2481121d <JS Function Module (SharedFunctionInfo 0x4141d2f1)>#3#,request=0x4143a8ad <String[12]: raspi-serial>,parent=0x248111d9 <a Module with map 0x2d80d2a9>#17#,isMain=0x57a08161 <false>)
   30: require [module.js:468] [pc=0x23946f98] (this=0x248111d9 <a Module with map 0x2d80d2a9>#17#,path=0x4143a8ad <String[12]: raspi-serial>)
   31: require(aka require) [internal/module.js:20] [pc=0x23946c6c] (this=0x57a080c9 <undefined>,path=0x4143a8ad <String[12]: raspi-serial>)
   32: /* anonymous */ [/home/pi/node_modules/raspi-io/lib/index.js:25] [pc=0x2394f230] (this=0x24811435 <an Object with map 0x5cd09de5>#18#,exports=0x24811435 <an Object with map 0x5cd09de5>#18#,require=0x24811261 <JS Function require (SharedFunctionInfo 0x414390d1)>#19#,module=0x248111d9 <a Module with map 0x2d80d2a9>#17#,__filename=0x248113fd <String[43]: /home/pi/node_modules/raspi-io/lib/index.js>,__dirname=0x248113e9 <String[34]: /home/pi/node_modules/raspi-io/lib>)
   33: _compile [module.js:541] [pc=0x239460bc] (this=0x248111d9 <a Module with map 0x2d80d2a9>#17#,content=0x24826e05 <Very long string[88405]>#20#,filename=0x248113fd <String[43]: /home/pi/node_modules/raspi-io/lib/index.js>)
   34: .js [module.js:550] [pc=0x2393de74] (this=0x24811189 <an Object with map 0x2d80cfbd>#1#,module=0x248111d9 <a Module with map 0x2d80d2a9>#17#,filename=0x248113fd <String[43]: /home/pi/node_modules/raspi-io/lib/index.js>)
   35: load [module.js:458] [pc=0x2393c4b8] (this=0x248111d9 <a Module with map 0x2d80d2a9>#17#,filename=0x248113fd <String[43]: /home/pi/node_modules/raspi-io/lib/index.js>)
   36: tryModuleLoad(aka tryModuleLoad) [module.js:417] [pc=0x2393bfa8] (this=0x57a080c9 <undefined>,module=0x248111d9 <a Module with map 0x2d80d2a9>#17#,filename=0x248113fd <String[43]: /home/pi/node_modules/raspi-io/lib/index.js>)
   37: _load [module.js:409] [pc=0x23936f94] (this=0x2481121d <JS Function Module (SharedFunctionInfo 0x4141d2f1)>#3#,request=0x41438c25 <String[8]: raspi-io>,parent=0x2483c779 <a Module with map 0x2d80d2a9>#21#,isMain=0x57a08161 <false>)
   38: require [module.js:468] [pc=0x23946f98] (this=0x2483c779 <a Module with map 0x2d80d2a9>#21#,path=0x41438c25 <String[8]: raspi-io>)
   39: require(aka require) [internal/module.js:20] [pc=0x23946c6c] (this=0x57a080c9 <undefined>,path=0x41438c25 <String[8]: raspi-io>)
   40: /* anonymous */ [/home/pi/blink.js:1] [pc=0x239464c4] (this=0x2483c875 <an Object with map 0x5cd09de5>#22#,exports=0x2483c875 <an Object with map 0x5cd09de5>#22#,require=0x2483c7dd <JS Function require (SharedFunctionInfo 0x414390d1)>#23#,module=0x2483c779 <a Module with map 0x2d80d2a9>#21#,__filename=0x2483c855 <String[17]: /home/pi/blink.js>,__dirname=0x2483c841 <String[8]: /home/pi>)
   41: _compile [module.js:541] [pc=0x239460bc] (this=0x2483c779 <a Module with map 0x2d80d2a9>#21#,content=0x2483ca21 <String[304]\: var raspi = require('raspi-io');\nvar five = require('johnny-five');\nvar board = new five.Board({\n  io: new raspi()\n});\n\nboard.on('ready', function() {\n\n  // Create an Led on pin 7 (GPIO4) on P1 and strobe it on/off\n  // Optionally set the speed; defaults to 100ms\n  (new five.Led('P1-7')).strobe();\n\n});\n>,filename=0x2483c855 <String[17]: /home/pi/blink.js>)
   42: .js [module.js:550] [pc=0x2393de74] (this=0x24811189 <an Object with map 0x2d80cfbd>#1#,module=0x2483c779 <a Module with map 0x2d80d2a9>#21#,filename=0x2483c855 <String[17]: /home/pi/blink.js>)
   43: load [module.js:458] [pc=0x2393c4b8] (this=0x2483c779 <a Module with map 0x2d80d2a9>#21#,filename=0x2483c855 <String[17]: /home/pi/blink.js>)
   44: tryModuleLoad(aka tryModuleLoad) [module.js:417] [pc=0x2393bfa8] (this=0x57a080c9 <undefined>,module=0x2483c779 <a Module with map 0x2d80d2a9>#21#,filename=0x2483c855 <String[17]: /home/pi/blink.js>)
   45: _load [module.js:409] [pc=0x23936f94] (this=0x2481121d <JS Function Module (SharedFunctionInfo 0x4141d2f1)>#3#,request=0x2483cb6d <String[17]: /home/pi/blink.js>,parent=0x57a08081 <null>,isMain=0x57a08129 <true>)
   46: /* anonymous */(aka /* anonymous */) [module.js:575] [pc=0x239369f8] (this=0x57a080c9 <undefined>)
   47: run(aka run) [bootstrap_node.js:352] [pc=0x2393683c] (this=0x57a080c9 <undefined>,entryFunction=0x41432ded <JS Function Module.runMain (SharedFunctionInfo 0x4141db61)>#24#)
   48: startup(aka startup) [bootstrap_node.js:144] [pc=0x3e542950] (this=0x57a080c9 <undefined>)
   49: /* anonymous */(aka /* anonymous */) [bootstrap_node.js:467] [pc=0x3e54060c] (this=0x57a08081 <null>,process=0x57a83b25 <a process with map 0x5cd0e8d5>#25#)
=====================
```

**Comment by nebrius** [comment]
Ah, sorry, I thought I had responded to this already. This is a known issue, however nothing is actually broken, this is just a warning (for now). It needs to be fixed, but we have at least until Node 7 before it will break, perhaps even longer.

**Comment by rbosneag** [comment]
I had the same question as the OP with the same code and similar setup (except I have node v6.6.0).

It's the first time I am using Johnny Five with RPi3 and got scared of the huge stack traces in JS and C. Is this the normal way of logging? 

The program then "hangs" as if in an infinite loop (maybe _event loop_?) but I thought I should be the one in control of what `node` does.

Also, if I am not using the serial port or other protocols is it possible to disable them when initializing the library? I could use the saved time for debugging, same with the console output.

Please excuse the noobness, I just got in touch with this world.

**Comment by kujukuju** [comment]
Getting this same error and then it hangs.

**Comment by RandScullard** [comment]
Any timeframe on a fix for this? I'm not getting the hanging behavior described by others, but the huge stack trace is certainly confusing for a newcomer like myself. It seems like this would just be a matter of upgrading to the latest version of serialport -- is that not correct?

**Comment by nebrius** [comment]
> Any timeframe on a fix for this? I'm not getting the hanging behavior described by others, but the huge stack trace is certainly confusing for a newcomer like myself. It seems like this would just be a matter of upgrading to the latest version of serialport -- is that not correct?

That sounds correct, yes. I didn't realize we'd had a major version bump in serialport. I should probably use greenkeeper or something.

**Comment by nebrius** [comment]
btw, I'm working on this, along with some other issues, and hope to have a new release of the raspi suite in the next week or so.

**Comment by nebrius** [comment]
version 2.0.0 of raspi-serial has been released with support for serialport 4, which removes this warning. It hasn't been integrated into raspi-io yet, there's a handful of changes I'd like to stage together, but it should be out soon.

**Comment by nebrius** [comment]
The changes have now been integrated and published with raspi-io 6.1.1. Let me know how it works!

**Comment by The-Alchemist** [comment]
@nebrius : everything good on my Pi:
- npm 3.10.8
- node 6.9.1

Also tried (successfullly) with node 7.0.0.
